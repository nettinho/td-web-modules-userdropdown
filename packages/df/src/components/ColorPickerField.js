import React from "react";
import PropTypes from "prop-types";
import { Form } from "semantic-ui-react";
import { GithubPicker } from "react-color";

export const ColorPickerField = ({
  label,
  name,
  required,
  value,
  onChange
}) => (
  <Form.Field required={required}>
    <label>{label}</label>
    <GithubPicker
      name="color_picker"
      value={value}
      onChangeComplete={color => {
        onChange(null, { name, value: color.hex });
      }}
    />
  </Form.Field>
);

ColorPickerField.propTypes = {
  label: PropTypes.string,
  name: PropTypes.string,
  required: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

export default ColorPickerField;
