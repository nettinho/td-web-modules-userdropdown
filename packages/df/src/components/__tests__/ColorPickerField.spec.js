import React from "react";
import { shallow } from "enzyme";
import { ColorPickerField } from "../ColorPickerField";

describe("<ColorPickerField />", () => {
  const label = "label";
  const name = "name";
  const required = true;
  const value = "Foo";
  const onChange = jest.fn();
  const props = { label, name, required, value, onChange };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<ColorPickerField {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("call onChange function", () => {
    const color = { hex: "hex_value" };
    const component = shallow(<ColorPickerField {...props} />);
    component.find({ name: "color_picker" }).simulate("changeComplete", color);
    expect(onChange).toBeCalledWith(null, { name: "name", value: "hex_value" });
  });
});
