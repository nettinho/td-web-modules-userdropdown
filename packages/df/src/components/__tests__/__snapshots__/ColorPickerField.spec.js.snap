// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorPickerField /> matches the latest snapshot 1`] = `
<FormField
  required={true}
>
  <label>
    label
  </label>
  <ColorPicker
    color={
      Object {
        "a": 1,
        "h": 250,
        "l": 0.2,
        "s": 0.5,
      }
    }
    colors={
      Array [
        "#B80000",
        "#DB3E00",
        "#FCCB00",
        "#008B02",
        "#006B76",
        "#1273DE",
        "#004DCF",
        "#5300EB",
        "#EB9694",
        "#FAD0C3",
        "#FEF3BD",
        "#C1E1C5",
        "#BEDADC",
        "#C4DEF6",
        "#BED3F3",
        "#D4C4FB",
      ]
    }
    name="color_picker"
    onChangeComplete={[Function]}
    triangle="top-left"
    value="Foo"
    width={200}
  />
</FormField>
`;
